sys/include/net/gnrc/gnrc_mac/gnrc_mac_types.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_mac  A common MAC type for providing key MAC parameters and helper functions
+ * @ingroup     net
+ * @brief       A common MAC type for providing key MAC parameters and helper functions.
+ * @{
+ *
+ * @file
+ * @brief       Internal types used by the GNRC_MAC entities
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_MAC_TYPES_H_
+#define GNRC_MAC_TYPES_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <kernel_types.h>
+#include <xtimer.h>
+#include <net/netdev2.h>
+#include <net/gnrc/netdev2.h>
+#include <net/gnrc/netif/hdr.h>
+#include <net/gnrc.h>
+#include <gnrc_mac.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/******************************************************************************/
+typedef struct {
+    uint8_t  addr[GNRC_NETIF_HDR_L2ADDR_MAX_LEN];
+    uint8_t  len;
+} l2_addr_t;
+#define GNRC_MAC_L2_ADDR_INIT      { {0}, 0 }
+
+/******************************************************************************/
+
+typedef enum {
+    TX_FEEDBACK_UNDEF = -1,
+    TX_FEEDBACK_SUCCESS,
+    TX_FEEDBACK_NOACK,
+    TX_FEEDBACK_BUSY
+} gnrc_mac_tx_feedback_t;
+
+/******************************************************************************/
+typedef struct gnrc_mac {
+
+
+    /* NETDEV device used by MAC */
+    gnrc_netdev2_t* gnrc_netdev2;
+    
+    /* Own address */
+    l2_addr_t l2_addr;
```

Comment:
With https://github.com/zhuoshuguo/RIOT/pull/2 you can pull this into `gnrc_netdev2_t`, too. However, I would suggest to not add an extra type for that. Just add the fields `l2_addr` (as a `byte-array`) and `l2_addr_len` (as a `byte`) into the `#ifdef MODULE_GNRC_MAC`'d section. Since `l2_addr_len` will be out of alignment I prefer it to be last in that type however.